t/lib/CorpusTest.pm

```diff
@@ -64,56 +104,150 @@ sub _validity_tests {
     local $ENV{BSON_EXTJSON_FORCE} = 1;
 
     my $bson_type = $json->{bson_type};
+    my $deprecated = $json->{deprecated};
 
     for my $case ( @{ $json->{valid} } ) {
-        local $Data::Dumper::Useqq = 1;
-
-        my $desc = $case->{description};
-        my $wrap = $bson_type =~ /\A(?:0x01|0x10|0x12)\z/;
-        my $codec = BSON->new( prefer_numeric => 1, wrap_numbers => $wrap, ordered => 1 );
-        my $lossy = $case->{lossy};
-
-        my $B = $case->{bson};
-        my $E = $case->{extjson}; # could be undef
-
-        my $cB = exists($case->{canonical_bson}) ? $case->{canonical_bson} : $B;
-        my $cE = exists($case->{canonical_extjson}) ? $case->{canonical_extjson} : $E;
-
-        my $skip_extjson = !(defined($E) && _extjson_ok($bson_type, $E));
-
-        $B = pack( "H*", $B );
-        $cB = pack( "H*", $cB );
-
-        $E = _normalize( $E, "$desc: normalizing E"  );
-        $cE = _normalize( $cE, "$desc: normalizing cE"  );
-
-        _bson_to_bson( $codec, $B, $cB, "$desc: B->cB" );
```

Comment:
Since the former versions of these used `bson` and `canonical_bson` I'm unsure how they would map to the new data fields. I've added a BSON roundtrip test for canonical/converted now, but I'm unsure if this is sufficient.